src/htmxl/compose/write/elements.py

```diff
@@ -72,9 +72,10 @@ def write_td(element, writer, styler, style):
     logger.debug("Writing <td> at {}".format(writer.ref))
     style = styler.get_style(element) or style
     with writer.record() as recording:
-        data_type = _type_map[element.attrs.get("data-type")]
-        logger.debug("Setting cell {} to {}".format(writer.ref, data_type))
-        write_value(data_type(element.text.strip()), writer, styler, style)
```

Comment:
importantly, i think this loses the `data_type` mapping. I think for now, you should maybe alter this so that if there's text do the data-type mapping. if not do the child iteration that you do here.

Long term, classes like `data-type` really should just be applied to all children automatically, but that will require more of a rework than necessary here.